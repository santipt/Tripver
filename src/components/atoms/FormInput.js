// Importing react utilities
import React from 'react';
import { Dimensions, StyleSheet, TextInput, Text, View } from 'react-native';

// Importing components
import * as Colors from '../../styles/colors';

// Getting screen dimensions
const { width, height } = Dimensions.get('screen');

export default function FormInput({ labelName, ...props }) {
  return (
    <View>
      {
        props.showLabel ?
          <Text style={{...styles.title, ...props.label_style}}>{labelName}</Text>
          : null
      }
      <TextInput
        label={labelName}
        style={{ ...styles.input, ...props.style }}
        placeholder={labelName}
        textContentType={props.textContentType}
        secureTextEntry={props.secureTextEntry}
        autoCompleteType={props.autoCompleteType}
        keyboardType={props.keyboardType}
        autoCapitalize={props.autoCapitalize}
        onChangeText={props.onChangeText}
        maxLength={props.maxLength}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  input: {
    marginTop: 10,
    marginBottom: 10,
    width: width / 1.2,
    height: 44,
    paddingVertical: 5,
    paddingHorizontal: 10,    
    borderRadius: 30,
    backgroundColor: 'white',
    fontSize: 15,
  },
  title:{
    marginLeft:10,
    fontWeight: 'bold',
    color: Colors.BLACK,
  },
});