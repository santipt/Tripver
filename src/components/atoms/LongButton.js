// Importing react utilities
import { useLinkProps } from '@react-navigation/native';
import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

// Importing components
import * as Colors from '../../styles/colors';

export default function LongButton({...props}) {
    return(
    <TouchableOpacity style={{...styles.button_container, ...props.style}} onPress={props.onPress}>
        <Text style={{...styles.button_text, ...props.textStyle}}>{props.title}</Text>
    </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    button_container: {
        backgroundColor: Colors.SECONDARY,
        borderRadius: 30,
        padding: 10,
        position: 'relative',
        width:'83%'
    },
    button_text: {
        textAlign: 'center',
        fontSize: 18,
        fontWeight: 'bold',
        color: 'white',
    },
}); 