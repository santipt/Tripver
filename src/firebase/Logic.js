import { db, firebase } from './index';

// Calculating age from date of birth
const getAge = (dateString) => {

    var today = new Date();
    var birthDate = new Date(dateString);
    var age = today.getFullYear() - birthDate.getFullYear();
    var m = today.getMonth() - birthDate.getMonth();
    if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
        age--;
    }
    return age;
}

async function uploadProfilePicture(uri, name) {
    const blob = await new Promise((resolve, reject) => {
        const xhr = new XMLHttpRequest();
        xhr.onload = function () {
            resolve(xhr.response);
        };
        xhr.onerror = function (e) {
            console.log(e);
            reject(new TypeError('Network request failed'));
        };
        xhr.responseType = 'blob';
        xhr.open('GET', uri, true);
        xhr.send(null);
    });

    const ref = firebase
        .storage()
        .ref()
        .child(name);
    const snapshot = await ref.put(blob);

    // We're done with the blob, close and release it
    blob.close();

    return await snapshot.ref.getDownloadURL();
}

export async function createUser(data) {

    console.log("Inserting user data into firestore...")

    var name;
    var email;
    var profile_picture;
    var age = getAge(data.birth_date);//getAge(data.birth_date.replace(/-/g, '/'));
    var birth_date = new Date(data.birth_date);//new Date(data.birth_date.replace(/-/g, '/'));

    try {

        if (data.googleData != null) {
            name = data.googleData.user.givenName;
            email = data.googleData.user.email;
            // Get google image with more quality
            profile_picture = data.googleData.user.photoUrl.replace("s96", "s400");
        } else {
            name = data.name;
            email = data.email;
            //profile_picture = await uploadProfilePicture(data.profile_picture, data.uid);
            profile_picture = "https://lh3.googleusercontent.com/a-/AOh14Gj32recPlk45teYg20KnAt3WZX8i8kql9LcUJiSdcg=s400-c";
        }

        var userData = {
            name: name,
            email: email,
            profile_picture: profile_picture,
            current_location: data.current_location,
            birth_date: birth_date,
            phone: data.phone,
            about_me: data.about_me,
            countries: data.countries,
            languages: data.languages,
            hobbies: data.hobbies,
            user_type: data.user_type,
            age: age,
            gender: data.gender,
        }

        // Inserting user data
        db.collection("users").doc(data.uid).set(userData)
            .catch((error) => {
                console.error("Error adding document: ", error);
            });

    } catch (err) {
        console.log(err);
        return err;
    }

};

export async function editUser(data) {

    // Checking if the profile picture is local
    // If it's not local then we don't update anything
    // If it's local we delete the old one and change it for the new one
    if(String(data.profile_picture).includes('googleusercontent') || String(data.profile_picture).includes('firebasestorage')){
        delete data.profile_picture;
    }else{
        // Delete old image in firestore storage

        // Update new image in firestore storage
    }

    console.log(data)

    var userId = firebase.auth().currentUser.uid;

    console.log("Updating user data... ", userId)
    
    db.collection('users').doc(userId).update(data).then(() => {
        console.log('User updated!');
        return;
    });
};

export function deleteUser() {

};

export async function getListOfUsers(currentUser) {

    var users = [];
    var data = await db.collection("users").get();

    //Current user
    //const {uid} = auth().currentUser;

    data.docs.forEach(item => {
        if (item.data().email != currentUser) {
            users.push(item.data())
        }
    })

    return users;
};

export async function userExists(email) {

    var exists = false;
    var data = await db.collection("users").get();

    data.docs.forEach(item => {
        if (email == item.data().email) {
            exists = true;
        }
    })

    return exists;
}


export function getUserData(user) {
    /* In order to know which user is logged in
        firebase.auth().onAuthStateChanged((user) => {
            if (user) {
              console.log('User uid: ', user.uid);
            }
          });*/
};
