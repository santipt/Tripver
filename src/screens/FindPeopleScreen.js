// Importing react utilities
import React, { useContext, useEffect, useState } from 'react';
import { StyleSheet, View, SafeAreaView } from 'react-native';

// Importing components
import { AuthContext } from '../navigation/AuthProvider';
import ListOfPeople from '../components/molecules/ListOfPeople'
import Loading from '../components/atoms/Loading';
import GlobalStyles from '../styles/GlobalStyles';


export default function FindPeopleScreen({ ...props }) {
  const { user, loading } = useContext(AuthContext);

  if (loading) {
    return <Loading />;
  }

  return (
    <SafeAreaView style={GlobalStyles.androidSafeArea} >
      <ListOfPeople user={user} userType={props.userType}></ListOfPeople>
    </SafeAreaView >
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});