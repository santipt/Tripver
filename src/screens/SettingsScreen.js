import React from 'react';
import { StyleSheet, View, SafeAreaView, Text, Button, TouchableOpacity } from 'react-native';


export default function EditProfileScreen({ navigation }) {
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text style={{ fontSize: 30 }}>Settings!</Text>
        <Button onPress={() => navigation.goBack()} title="Dismiss" />
      </View>
    );
  }