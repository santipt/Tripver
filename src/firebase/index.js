import * as firebase from 'firebase';
import '@firebase/auth';
import '@firebase/firestore';

// Clients ids for google sign in
const IOS_CLIENT_ID = "258735134861-ofh5l2autr27ds68hukkp823l44un220.apps.googleusercontent.com";
const ANDROID_CLIENT_ID= "258735134861-apeesi2b2tneaqmkocr0f24hfknnk7i2.apps.googleusercontent.com"

var firebaseConfig = {
  apiKey: "AIzaSyANwjqxLqpNAQK6PcxrNNfoZk-fC8T3-K8",
  authDomain: "tripver-d5fa9.firebaseapp.com",
  projectId: "tripver-d5fa9",
  storageBucket: "tripver-d5fa9.appspot.com",
  messagingSenderId: "1028452737839",
  appId: "1:1028452737839:web:832eab197c063de8da77da",
  measurementId: "G-WHZJEGD5CB"
};

if (!firebase.apps.length) {
  firebase.initializeApp(firebaseConfig);
}

// Initialazing cloud firestore
var db = firebase.firestore();

export { firebase, db, IOS_CLIENT_ID, ANDROID_CLIENT_ID  };

// Example data base insert
/*db.collection("users").add({
            name: googleUser.user.givenName,
            email: googleUser.user.email,
            profile_picture: googleUser.user.photoUrl,
            about_me: "adsffafa",
            current_location: "Valencia",
            birth_date:"12/12/10",
            countries:["adsf","fsda"],
            languages:["fsafd","gfdsg"],
            user_type:"local",
        })
        .catch((error) => {
            console.error("Error adding document: ", error);
        });*/