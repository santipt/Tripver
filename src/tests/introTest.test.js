import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount, simulate } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import { waitFor } from 'react-native-testing-library';

import LoginScreen from '../screens/LoginScreen';
import SignUpScreen from '../screens/SignUpScreens/SignUpScreen';
import SignUpScreen1 from '../screens/SignUpScreens/SignUpScreen1';
import SettingsScreen from '../screens/SettingsScreen';
import ProfileScreen from '../screens/ProfileScreen';
import FindPeopleScreen from '../screens/FindPeopleScreen';


import FormInput from '../components/atoms/FormInput'
import LongButton from '../components/atoms/LongButton'

import { getAge, getListOfLocals, getListOfTripvers, changeEmail, changePassword, userExists, calculateDistance } from '../firebase/Logic';

Enzyme.configure({ adapter: new Adapter() });

// I have to do 25 tests:
// 6/25


// --------------------------------------------------------
// ----------------- TESTING RENDERING --------------------
// --------------------------------------------------------

// Testing if the SignUpScreen renders correctly
test('Renders correctly SignUpScreen', () => {
  const tree = renderer.create(<SignUpScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});


// --------------------------------------------------------
// ----------------- TESTING LOGIC FUNCTIONS --------------
// --------------------------------------------------------
describe("Testing logic functions", () => {

  // Testing GetListOfLocals() function
  test('Testing GetListOfLocals() function', async (done) => {

    // Passing an existing email (the current user) in order to avoid to show on the list
    getListOfLocals('julia@hotmail.es').then((data) => {
      expect(data).toEqual(
        expect.arrayContaining([
          // Expecting two users (in total are 3)
          expect.objectContaining({ id: 'KalIq4IRH8Ozr2T2g5IW0E7SgJn1' }),
          expect.objectContaining({ id: 'opkVgnmqmlZxZQC2bMisPV5fzLo2' })
        ]));
      done();
    });

  });

  // Testing GetListOfTripvers() function
  test('Testing GetListOfTripvers() function', async (done) => {

    // Passing an existing email (the current user) in order to avoid to show on the list
    getListOfTripvers('ramon@hotmail.es').then((data) => {
      expect(data).toEqual(
        expect.arrayContaining([
          // Expecting one user (in total are 3)
          expect.objectContaining({ id: 'FXjIr88P3ThDDoaaJQYfMDKxNL02' }),
        ]));
      done();
    });

  });


  // Testing if the useExists() works correctly with a user that doesn't exists
  test('Testing userExists() function', (done) => {
    // manolo123@gmail.com doesn't exists in the db
    userExists('manolo123@gmail.com').then((data) => {
      expect(data).toEqual(false);
      done();
    })
  });

  // // Testing changeEmail() function
  // test('Testing changeEmail() function', async (done) => {

  //   // Passing an the new email, existing email, and the wrong password
  //   changeEmail('ramon1@hotmail.es', 'ramon@hotmail.es', '1234').catch((data) => {
  //     console.log(data.message)
  //     expect(data.message).toEqual('The password is invalid or the user does not have a password.');
  //     done();
  //   });

  // });

  // // Testing changePassword() function
  // test('Testing changePassword() function', async (done) => {

  //   // Passing an existing email, the wrong current password and the new password
  //   changePassword('ramon@hotmail.es', '1234', '12345678').catch((data) => {
  //     console.log(data)
  //     expect(data.message).toEqual('The password is invalid or the user does not have a password.');
  //     done();
  //   });

  // });

  // Testing if the getAge() function works correctly
  test('Testing getAge() function', () => {
    // Giving a date and expecting the correct age of the date
    expect(getAge('11/10/1999')).toEqual(21)
  })

  // Testing if the calculateDistance() function works correctly
  test('Testing calculateDistance() function', () => {
    // Giving two coord and expecting the right distance
    expect(calculateDistance(51.12382554627786, 17.044555543152967, 51.11019424353001, 17.034513353317728)).toEqual(1.7)
  })

});

// --------------------------------------------------------
// ----------------- TESTING USER BEHAVIOUR ---------------
// --------------------------------------------------------
describe("LogIn form behaviour", () => {
  test('Renders correctly LoginScreen', () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('validate user inputs', async () => {

    let inst;
    inst = renderer.create(<LoginScreen />)

    await waitFor(() => {
      const button = inst.root.findByType(LongButton);
      const inputEmail = inst.root.findAllByType(FormInput)[0];
      const inputPassword = inst.root.findAllByType(FormInput)[1];

      inputEmail.props.onChangeText('julia@hotmail.es');
      inputPassword.props.onChangeText('12345678');

      // button.props.onPress();

      // expect(button).toBeTruthy();

      expect(inputEmail.props.value).toBe('julia@hotmail.es');
      expect(inputPassword.props.value).toBe('12345678');
    });

  })
});

describe("SignUp form behaviour", () => {
  test('Renders correctly LoginScreen', () => {
    const tree = renderer.create(<SignUpScreen1 />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('validate user inputs', async () => {

    let inst;
    inst = renderer.create(<SignUpScreen1 />)

    await waitFor(() => {
      const inputName = inst.root.findAllByType(FormInput)[0];
      const inputEmail = inst.root.findAllByType(FormInput)[1];
      const inputPassword = inst.root.findAllByType(FormInput)[2];

      inputName.props.onChangeText('Julia');
      inputEmail.props.onChangeText('julia@hotmail.es');
      inputPassword.props.onChangeText('12345678');

      // button.props.onPress();

      // expect(button).toBeTruthy();

      expect(inputName.props.value).toBe('Julia');
      expect(inputEmail.props.value).toBe('julia@hotmail.es');
      expect(inputPassword.props.value).toBe('12345678');
    });

  })
});

