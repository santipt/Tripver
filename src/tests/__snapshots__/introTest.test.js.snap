// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn form behaviour Renders correctly LoginScreen 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEnabled={false}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#e6e6e6",
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginRight": 15,
                "marginTop": 15,
                "textAlign": "center",
              }
            }
          >
            Sign up
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": "20%",
                "width": "32%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "500",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    Object {
                      "fontSize": 24,
                      "fontWeight": "bold",
                      "marginBottom": 30,
                      "marginTop": 20,
                    },
                  ],
                ],
              ]
            }
          >
            Tripver
          </Text>
          <View>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCompleteType="email"
              keyboardType="email-address"
              label="Email"
              onChangeText={[Function]}
              placeholder="Email"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 30,
                  "fontSize": 15,
                  "height": 44,
                  "marginBottom": 10,
                  "marginTop": 10,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                  "width": 625,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View>
            <TextInput
              allowFontScaling={true}
              label="Password"
              onChangeText={[Function]}
              placeholder="Password"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 30,
                  "fontSize": 15,
                  "height": 44,
                  "marginBottom": 10,
                  "marginTop": 10,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                  "width": 625,
                }
              }
              textContentType="password"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#f0b65f",
                "borderRadius": 30,
                "marginTop": 30,
                "opacity": 1,
                "padding": 10,
                "position": "relative",
                "width": "83%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Log In
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginTop": 30,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 1,
                  "width": "35%",
                }
              }
            />
            <View>
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 10,
                    "textAlign": "center",
                    "width": 50,
                  }
                }
              >
                OR
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 1,
                  "width": "35%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 30,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 8,
                "marginTop": 30,
                "opacity": 1,
                "padding": 10,
                "width": "50%",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": "120%",
                  "marginLeft": 0,
                  "marginTop": -2,
                  "width": "15%",
                }
              }
            />
            <Text
              style={
                Object {
                  "alignItems": "center",
                  "color": "#000000",
                  "flex": 1,
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginLeft": 15,
                  "marginRight": 35,
                  "textAlign": "center",
                }
              }
            >
              Google
            </Text>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`Renders correctly SignUpScreen 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEnabled={false}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#e6e6e6",
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginRight": 15,
                "marginTop": 15,
                "textAlign": "center",
              }
            }
          >
            Log In
          </Text>
        </View>
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": "20%",
              "marginBottom": 40,
              "marginLeft": 30,
              "marginTop": 0,
              "width": "34%",
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  Object {
                    "fontSize": 30,
                    "fontWeight": "bold",
                    "marginBottom": 80,
                    "marginLeft": 40,
                  },
                ],
              ],
            ]
          }
        >
          Welcome to Tripver
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#f0b65f",
                "borderRadius": 30,
                "opacity": 1,
                "padding": 10,
                "position": "relative",
                "width": "83%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Create account
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 30,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 8,
                "marginTop": 30,
                "opacity": 1,
                "padding": 10,
                "width": "83%",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": "110%",
                  "marginLeft": 0,
                  "marginTop": -1,
                  "width": "8%",
                }
              }
            />
            <Text
              style={
                Object {
                  "alignItems": "center",
                  "color": "#000000",
                  "flex": 1,
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginLeft": 15,
                  "marginRight": 35,
                  "textAlign": "center",
                }
              }
            >
              Continue with Google
            </Text>
          </View>
          <Text
            style={
              Object {
                "bottom": 0,
                "color": "#FFFFFF",
                "fontSize": 12,
                "left": 0,
                "margin": 20,
                "marginBottom": 50,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            By typing Continue or Create Account, I agree to Tripver's 
            <Text
              onPress={[Function]}
              style={
                Object {
                  "textDecorationLine": "underline",
                }
              }
            >
              Terms of Service, Privacy Policy and Nondicrimination Policy.
            </Text>
          </Text>
          <View
            style={
              Object {
                "bottom": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "left": 0,
                "marginBottom": 20,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                }
              }
            >
              In colaboration with
            </Text>
            <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": "150%",
                  "width": "28%",
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`SignUp form behaviour Renders correctly LoginScreen 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEnabled={false}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#e6e6e6",
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <Text />
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
                "marginBottom": 60,
                "marginTop": -10,
              }
            }
          >
            Let's get started!
          </Text>
          <View>
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                }
              }
            >
              Name
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCompleteType="name"
              keyboardType="default"
              label="Name"
              maxLength={16}
              onChangeText={[Function]}
              placeholder="Name"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 30,
                  "fontSize": 15,
                  "height": 44,
                  "marginBottom": 30,
                  "marginTop": 10,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                  "width": 625,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                }
              }
            >
              Email
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCompleteType="email"
              keyboardType="email-address"
              label="Email"
              onChangeText={[Function]}
              placeholder="Email"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 30,
                  "fontSize": 15,
                  "height": 44,
                  "marginBottom": 30,
                  "marginTop": 10,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                  "width": 625,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                }
              }
            >
              Password
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCompleteType="password"
              label="Password"
              onChangeText={[Function]}
              placeholder="Password"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 30,
                  "fontSize": 15,
                  "height": 44,
                  "marginBottom": 30,
                  "marginTop": 10,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                  "width": 625,
                }
              }
              textContentType="password"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#f0b65f",
                "borderRadius": 30,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 8,
                "marginTop": 30,
                "opacity": 1,
                "padding": 10,
                "width": "50%",
              }
            }
          >
            <Text
              style={
                Object {
                  "alignItems": "center",
                  "color": "#FFFFFF",
                  "flex": 1,
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginLeft": 15,
                  "textAlign": "center",
                }
              }
            >
              Next
            </Text>
            <Text />
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;
